src/foam/dao/CSVSink.js

```diff
@@ -76,34 +101,97 @@ foam.CLASS({
             Null: function(value) {}
           }, function(value) {
             this.output_(value.toString());
-          })
+        }),
+      javaCode: `
+        StringBuilder sb = new StringBuilder();
+        sb.append(getCsv());
+        String s = value.toString();
+
+        if (s.indexOf("\\"") != -1 || s.indexOf(",") != -1) {
```

Comment:
You might want to use logic similar to the code block above to keep js and java behavior more consistent with each other. You can't use an mmethod but you can use if/else similar to the java json outputter: https://github.com/foam-framework/foam2/blob/master/src/foam/lib/json/Outputter.java#L213